internal/querycompute/querycompute.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+
+option go_package = "github.com/Microsoft/hcsshim/internal/querycompute;querycompute";
+
+import weak "gogoproto/gogo.proto";
+
+service QueryCompute {
```

Comment:
I feel maybe we can improve the naming here. The service is named `QueryCompute` but the input parameter for the request is a `container_id`. Also, if we add other requests to this service they might not be limited to compute queries.  If we only expect to use this for querying UVM's then maybe we should name the service `QueryPod` and name the request parameter `pod_id`. If this is going to be a service that will work for all containers/tasks then maybe we should name it something like `ExtendedTask`. What do you think?

Reply:
I don't think we have a good idea of what else may be added in the future, so to be on the safe side I'd like to keep it open to potentially work for all containers/tasks. I'd be fine calling it `ExtendedTask` in that case if that's preferred.

Reply:
Sounds good.

Reply:
@kevpar Any thoughts on this? Otherwise I'll change to `ExtendedTask` shortly

Reply:
Is the idea that this would be a catchall service to host any other API we want to add that is not present in containerd's `Task` service? If so I think that's okay. This should all be internal only anyways, so we can change it later.

Reply:
For right now the intention is only to be used with ncproxy but since we don't know what that might evolve to, we may want it to be a catchall service as you described. I'll change it since it's not a lot of code anyways.